gr-iio/lib/device_sink_impl.cc

```diff
@@ -116,6 +117,11 @@ device_sink_impl::device_sink_impl(iio_context* ctx,
 
             iio_channel_enable(chn);
             channel_list.push_back(chn);
+
+            const iio_data_format *data = iio_channel_get_data_format(chn);
+            int bit_length = (data->length + (8 - 1)) / 8;
+            signature.push_back(bit_length);
+
```

Comment:
please avoid duplicating code. This could very well be a small `constexpr unsigned int bit_length(const iio_channel* data){…}` function in a private header in the lib/ directory. See how the same implementation in `device_source_impl` is better commented! It would be an advantage if that was the code I've read before reading this. But I read this first and scratched my head.

Generally, I'd say this applies to this whole `for` loop, but we're not going to force you to refactor this whole set of constructors now.